### PR TITLE
Update cache plugin documentation

### LINKED ISSUE #63616: Inventory cache settings 2.7.13
##### SUMMARY

According to the documentation, if _cache_plugin_ and _cache_connection_ are not set in _[inventory]_ section of ansible.cfg and cache set to True, the values should be taken from fact_caching and fact_caching_connection in defaults section.
This is working with ansible 2.8.5 but returns an error in 2.7.13

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Inventory cache settings

##### ANSIBLE VERSION

```paste below
ansible 2.7.13
  config file = /home/xxx/automation-cisco/ansible/ansible.cfg
  configured module search path = ['/home/xxx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Oct  7 2019, 12:59:55) [GCC 8.3.0]
```

##### CONFIGURATION

```paste below
CACHE_PLUGIN(/home/xxx/automation-cisco/ansible/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/home/xxx/automation-cisco/ansible/ansible.cfg) = cache
```
The config in [inventory] section doesn't show in this output (no matter if cache_plugin is set or not) - OK found #46097

In Ansible 2.8.5 :
```
INVENTORY_CACHE_ENABLED(/home/xxx/automation-cisco/ansible/ansible.cfg) = True
```

##### STEPS TO REPRODUCE

Set the following in ansible.cfg

```
[defaults]
fact_caching = jsonfile
fact_caching_connection = cache

[inventory]
cache = True
#linked to fact_caching options if not specified (OK in 2.8+, not in 2.7)
#cache_plugin = jsonfile
#cache_connection = cache
```
Will return an error; uncomment cache_plugin AND cache_connection and it will work

```bash
ansible-inventory --graph
```

##### EXPECTED RESULTS

The inventory cache options should inherit from fact_caching options, even the error messages states it.
https://docs.ansible.com/ansible/2.7/plugins/cache.html
> If an inventory-specific cache plugin is not provided and inventory caching is enabled, the fact cache plugin is used for inventory.

##### ACTUAL RESULTS

```paste below
 [WARNING]:  * Failed to parse /home/xxx/automation-cisco/ansible/inventory/01-netbox.yml with auto plugin: error, 'None' inventory cache plugin requires the one of the following to be set: ansible.cfg: [default]: fact_caching_connection, [inventory]: cache_connection;
Environment: ANSIBLE_INVENTORY_CACHE_CONNECTION, ANSIBLE_CACHE_PLUGIN_CONNECTION.to be set to a writeable directory path
```

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #63616

Since 2.8 cache plugins using BaseCacheModule and BaseFileCacheModule as the base class can be used for inventory.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/plugins/cache.rst

### CODE REVIEW

File: docs/docsite/rst/plugins/cache.rst [hunk 1]
```diff
@@ -50,8 +50,9 @@ A custom cache plugin is enabled by dropping it into a ``cache_plugins`` directo
 Enabling Inventory Cache Plugins
 --------------------------------
 
-Inventory may be cached using a file-based cache plugin (like jsonfile). Check the specific inventory plugin to see if it supports caching. Cache plugins inside a collection are not supported for caching inventory.
-If an inventory-specific cache plugin is not specified Ansible will fall back to caching inventory with the fact cache plugin options.
+Prior to Ansible 2.8, inventory plugins that support caching could use file-based cache plugins, such as jsonfile, pickle, and yaml. For Ansible 2.8 and later all cache plugins shipped with Ansible are supported. Check the specific inventory plugin to see if it supports caching. To cache inventory with a custom plugin in your plugin path, take a look at the developer's guide for cache plugins. Cache plugins inside a collection are not supported for caching inventory.
```
Comment (Akasurde) [inline docs/docsite/rst/plugins/cache.rst]:
```suggestion
Prior to Ansible 2.8, inventory plugins that support caching could use file-based cache plugins, such as jsonfile, pickle, and yaml. For Ansible 2.8 and later all cache plugins shipped with Ansible are supported. Check the specific inventory plugin to see if it supports caching. To cache inventory with a custom plugin in your plugin path, take a look at the `developer's guide for cache plugins <https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#cache-plugins>`_. Cache plugins inside a collection are not supported for caching inventory.
```

Comment (s-hertel) [reply docs/docsite/rst/plugins/cache.rst]: Thanks :)

**Review by Akasurde — APPROVED** [review]
LGTM

File: docs/docsite/rst/plugins/cache.rst [hunk 2]
```diff
@@ -50,8 +50,9 @@ A custom cache plugin is enabled by dropping it into a ``cache_plugins`` directo
 Enabling Inventory Cache Plugins
 --------------------------------
 
-Inventory may be cached using a file-based cache plugin (like jsonfile). Check the specific inventory plugin to see if it supports caching. Cache plugins inside a collection are not supported for caching inventory.
-If an inventory-specific cache plugin is not specified Ansible will fall back to caching inventory with the fact cache plugin options.
+Prior to Ansible 2.8, inventory plugins that support caching could use file-based cache plugins, such as jsonfile, pickle, and yaml. For Ansible 2.8 and later all cache plugins shipped with Ansible are supported. Check the specific inventory plugin to see if it supports caching. To cache inventory with a custom plugin in your plugin path, take a look at the `developer's guide for cache plugins <https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#cache-plugins>`_. Cache plugins inside a collection are not supported for caching inventory.
```
Comment (gundalow) [inline docs/docsite/rst/plugins/cache.rst]:
```suggestion
Prior to Ansible 2.8, inventory plugins that support caching could use file-based cache plugins, such as jsonfile, pickle, and yaml. For Ansible 2.8 and later all cache plugins shipped with Ansible are supported. Check the specific inventory plugin to see if it supports caching. To cache inventory with a custom plugin in your plugin path, take a look at the :ref:`developer guide on cache plugins<developing_cache_plugins>`. Cache plugins inside a collection are not supported for caching inventory.
```

**Review by gundalow — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/plugins/cache.rst (hunk 2 above)
Comment (s-hertel) [reply docs/docsite/rst/plugins/cache.rst]: Thanks, links are difficult :-)

**Review by gundalow — APPROVED** [review]

**Comment by s-hertel** [comment]
Thanks for the reviews. @acozine let me know at the beginning of this week "I might make a few changes on top of your changes" in regard to this PR, so this is just waiting on that.

**Pull request merged**